### PR TITLE
Remove code guarded by #ifdef kernel macros

### PR DESCRIPTION
**Pull request by tnek**
This should be the last remaining bits of the kernel extension in our codebase

Linked bug: https://github.com/google/santa/issues/738

### CODE REVIEW

File: Source/common/SNTCommon.h [hunk 1]
```diff
@@ -16,40 +16,16 @@
...
-  kSantaUserClientCheckCache,
-  kSantaUserClientCacheBucketCount,
-  kSantaUserClientFilemodPrefixFilterAdd,
-  kSantaUserClientFilemodPrefixFilterReset,
-
-  // Any methods supported by the driver should be added above this line to
-  // ensure this remains the count of methods.
-  kSantaUserClientNMethods,
-};
-
```
Comment (russellhancox) [inline Source/common/SNTCommon.h:52]: The `santa_queuetype_t` enum below can be removed too

**Review by russellhancox — CHANGES_REQUESTED** [review]

File: Source/common/SNTLogging.h [hunk 2]
```diff
@@ -19,21 +19,6 @@
 #ifndef SANTA__COMMON__LOGGING_H
```
[the remark this answers is not in the document]
Comment (tnek) [reply Source/common/SNTLogging.h:19]: Done

File: Source/common/SNTCommon.h (hunk 1 above)
Comment (tnek) [reply Source/common/SNTCommon.h:52]: Done

File: Source/common/SantaCache.h [hunk 3]
```diff
@@ -41,7 +38,6 @@
 #define OSDecrementAtomic(addr) OSAtomicDecrement64((volatile int64_t *)addr)
```
[the remark this answers is not in the document]
Comment (tnek) [reply Source/common/SantaCache.h]: Done

**Review by russellhancox — APPROVED** [review]

**Pull request merged**